### PR TITLE
[Text Generation] Avoid mutating the original logits in-place when mapping from logits to token.

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Feature Description

`TokenGenerator.generate(logits: numpy.ndarray)` takes a logits array and, if `do_sample=True`, optionally mutates them to enforce the appropriate sampling strategy. 

This results in the correct generation of tokens, but an in-place modification of the logits array. Logits array is then returned to the user in the mutated form. This can be confusing for the users who are interested in logits value, specially when returned together with prompt logits:

image
First column "before": the generated logits returned to the user are mutated (the distribution is spikier, arguably close to Dirac distribution)
Second column "after": the generated logits are the non-mutatated, original logits, consistent with the prompt logits.

### CODE REVIEW

File: src/deepsparse/transformers/utils/token_generator.py [hunk 1]
```diff
@@ -77,6 +77,10 @@ def generate(self, logits: numpy.ndarray) -> numpy.ndarray:
         :param logits: the logits from the model with shape (vocab_size,)
         :return: the sampled token
         """
+        # make a copy of logits to avoid modifying the original
+        # logits distribution in-place
+        logits = logits.copy()
```
Comment (mgoin) [inline src/deepsparse/transformers/utils/token_generator.py:82]: does this only happen when we are explicitly returning logits? this seems like it could be expensive

Comment (dbogunowicz) [reply src/deepsparse/transformers/utils/token_generator.py:82]: Good point. This will happen regardless of whether we are returning logits to the user or not. However, I do not think that it is worth making this logic more complex in favor of memory benefits. Note that those are not the logits for the full sequence being copied, but only the last logits array of shape `vocab_size`.

**Review by mgoin — APPROVED** [review]
